cpp/include/raft/core/mdspan.hpp

```diff
@@ -222,4 +222,112 @@ RAFT_INLINE_FUNCTION auto unravel_index(Idx idx,
   }
 }
 
-}  // namespace raft
\ No newline at end of file
+template <class ElementType, class Extents, class Layout, class Accessor>
+constexpr bool is_row_or_column_major(mdspan<ElementType, Extents, Layout, Accessor> /* m */)
+{
+  return false;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_row_or_column_major(mdspan<ElementType, Extents, layout_left, Accessor> /* m */)
+{
+  return true;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_row_or_column_major(mdspan<ElementType, Extents, layout_right, Accessor> /* m */)
+{
+  return true;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_row_or_column_major(mdspan<ElementType, Extents, layout_stride, Accessor> m)
+{
+  return m.is_exhaustive();
+}
+
+template <class ElementType, class Extents, class Layout, class Accessor>
+constexpr bool is_row_major(mdspan<ElementType, Extents, Layout, Accessor> /* m */)
+{
+  return false;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_row_major(mdspan<ElementType, Extents, layout_left, Accessor> /* m */)
+{
+  return false;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_row_major(mdspan<ElementType, Extents, layout_right, Accessor> /* m */)
+{
+  return true;
+}
+
+template <class ElementType, class Extents, class Accessor>
+constexpr bool is_row_major(mdspan<ElementType, Extents, layout_stride, Accessor> m)
+{
+  return m.is_exhaustive();
```

Comment:
[required] I don't think this is quite right.  Counterexample: `layout_stride::mapping{1, num_rows}` (which is column major).  This should be straightforward to fix, though.

```c++
return m.is_exhaustive() && m.stride(1) == typename Extents::index_type(1);
```